src/workerd/io/worker.c++

```diff
@@ -2705,33 +2598,38 @@ void Worker::Actor::ensureConstructed(IoContext& context) {
         e.setDescription(kj::mv(description));
       }
 
-      impl->constructorFailedPaf.fulfiller->reject(kj::cp(e));
-      impl->classInstance = kj::mv(e);
+      impl.constructorFailedPaf.fulfiller->reject(kj::cp(e));
+      impl.classInstance = kj::mv(e);
     }));
 
-    impl->classInstance = Impl::Initializing();
+    impl.classInstance = ActorImpl::Initializing();
   }
 }
 
+// // TODO(now): We need to destroy Worker::ActorImpl under lock now
```

Comment:
Nit: TODO(now) here also. With the commented out block still here, it's not clear what action needs to be taken with this todo.